### ISSUE TITLE
need something that removes environments from functions

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr3misc#35]
I want to save an anonymous function in an object and serialize that object, but that object should not grow huge. It is then necessary that the function's environment hierarchy doesn't contain too much other stuff.

## Usecase:

```r
x = 1
large_obj = [huge object that takes lots of memory]
want_to_serialize = function(arg) return(arg + x)
saveRDS(want_to_serialize, "savefile.rds")
```
`want_to_serialize`'s environment now contains `large_obj` and the savefile grows large. Instead I would want a function `detachEnv(fun, keep, basis)`:
```r
want_to_serialize_lean = detachEnv(want_to_serialize, keep = "x")
saveRDS(want_to_serialize_lean, "savefile.rds")
```
code for this:
```r
# removes the environment from fun and potentially wraps it in a new environment
detachEnv <- function(fun, keep = character(0), basis = topenv(parent.frame())) {
  assertEnvironment(basis)
  assertCharacter(keep, any.missing = FALSE)
  assertFunction(fun)
  if (length(keep)) {
    keepvals <- mget(keep, parent.frame(), inherits = TRUE)
    basis <- new.env(parent = basis, size = length(keepvals))
    mapply(assign, names(keepvals), keepvals, MoreArgs = list(envir = basis))
  }
  environment(fun) <- basis
  fun
}
```
name up for debate

**Comment by mllg** [comment]
https://github.com/r-lib/carrier/

Something like this?

**Comment by berndbischl** [comment]
i think the do this in tidymodels too

**Comment by mb706** [comment]
this, only without the tidyverse awfulness

**Comment by mllg** [comment]
Closed in #50 ?
